lang/src/main/kotlin/org/partiql/lang/eval/BagOp.kt

```diff
@@ -70,7 +70,7 @@ fun interface ExprValueBagOp {
  */
 internal fun ExprValue.coerceToBag(): Sequence<ExprValue> = when {
     isUnknown() -> emptySequence()
-    this is StructExprValue || this is Scalar -> sequenceOf(this)
+    type === ExprValueType.STRUCT || type.isScalar -> sequenceOf(this)
```

Comment:
The original way to check a scalar depends on certain implementation. Scalar `ExprValue` does not have to extend `Scalar` interface. So for better maintainability, we use its type to check if it is a scalar.

Reply:
To be more specific, the implementation of scalar `ExprValue` does not depend on `Scalar` interface.